### PR TITLE
Fix JSON-LD object in the spec

### PR DESCRIPTION
**Pull request by gjgd**
Running jsonld-checker.com on the spec files reveals that a couple of JSON-LD objects in the examples are invalid. See https://www.jsonld-checker.com/?tab=1&url=https%3A%2F%2Fgithub.com%2Fw3c%2Fdid-core%2Fblob%2Fmaster%2Findex.html

Namely:
- [The UserHubEndpoint object](https://www.jsonld-checker.com/?json=%257B%250A%2520%2520%2522%2540context%2522%253A%2520%2522https%253A%252F%252Fschema.identity.foundation%252Fhub%2522%252C%250A%2520%2520%2522type%2522%253A%2520%2522UserHubEndpoint%2522%252C%250A%2520%2520%2522instances%2522%253A%2520%255B%250A%2520%2520%2520%2520%2522did%253Aexample%253A456%2522%252C%250A%2520%2520%2520%2520%2522did%253Aexample%253A789%2522%250A%2520%2520%255D%250A%257D&tab=0) which context (https://schema.identity.foundation/hub) is not a resolvable URL. We should either remove this object or fix it by adding a valid context. I wasn't able to find a context with these properties, **suggestions are welcome**.
- [A did document](https://www.jsonld-checker.com/?json=%257B%250A%2520%2520%2522%2540context%2522%253A%2520%2522https%253A%252F%252Fwww.w3.org%252Fns%252Fdid%252Fv1%2522%252C%250A%2520%2520%2522authentication%2522%253A%2520%255B%250A%2520%2520%2520%2520%2522did%253Aexample%253A12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew%253Bkey-id%253Dbafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%2522%250A%2520%2520%255D%252C%250A%2520%2520%2522created%2522%253A%2520%25222018-12-01T03%253A00%253A00Z%2522%252C%250A%2520%2520%2522id%2522%253A%2520%2522did%253Aexample%253A12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew%2522%252C%250A%2520%2520%2522proof%2522%253A%2520%257B%250A%2520%2520%2520%2520%2522created%2522%253A%2520%25222020-05-01T03%253A00%253A02Z%2522%252C%250A%2520%2520%2520%2520%2522creator%2522%253A%2520%2522did%253Aexample%253A12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew%253B%2520example%253Akey%253Did%253Dbafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%2522%252C%250A%2520%2520%2520%2520%2522signatureValue%2522%253A%2520%2522o9r6LxgoGN8FoaeeUA6EdDcv12GvDzFEmCgjWzvpur2YSQyA8W2r0SSWUK%252BnH5tMqzaFLun6wwZ1Eot37amGDg%253D%253D%2522%252C%250A%2520%2520%2520%2520%2522type%2522%253A%2520%2522ed25519Signature2018%2522%250A%2520%2520%257D%252C%250A%2520%2520%2522verificationMethod%2522%253A%2520%255B%250A%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522curve%2522%253A%2520%2522ed25519%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522expires%2522%253A%2520%25222019-12-01T03%253A00%253A00Z%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522id%2522%253A%2520%2522bafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%253Bexample%253Akey%253Didbafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522publicKeyBase64%2522%253A%2520%2522qmz7tpLNKKKdl7cD7PbejDiBVp7ONpmZbfmc7cEK9mg%253D%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522type%2522%253A%2520%2522EdDsaPublicKey%2522%250A%2520%2520%2520%2520%257D%250A%2520%2520%255D%252C%250A%2520%2520%2522updated%2522%253A%2520%25222020-05-01T03%253A00%253A00Z%2522%250A%257D&tab=0) which is missing the "proof" property in its context. This PR adds the security v2 context to this object [so that it becomes valid](https://www.jsonld-checker.com/?json=%257B%2520%2522%2540context%2522%253A%2520%255B%250A%2520%2520%2520%2520%2522https%253A%252F%252Fwww.w3.org%252Fns%252Fdid%252Fv1%2522%252C%250A%2520%2520%2520%2520%2522https%253A%252F%252Fw3id.org%252Fsecurity%252Fv2%2522%250A%2520%2520%255D%252C%250A%2520%2520%2522authentication%2522%253A%2520%255B%250A%2520%2520%2520%2520%2522did%253Aexample%253A12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew%253Bkey-id%253Dbafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%2522%250A%2520%2520%255D%252C%250A%2520%2520%2522created%2522%253A%2520%25222018-12-01T03%253A00%253A00Z%2522%252C%250A%2520%2520%2522id%2522%253A%2520%2522did%253Aexample%253A12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew%2522%252C%250A%2520%2520%2522proof%2522%253A%2520%257B%250A%2520%2520%2520%2520%2522created%2522%253A%2520%25222020-05-01T03%253A00%253A02Z%2522%252C%250A%2520%2520%2520%2520%2522creator%2522%253A%2520%2522did%253Aexample%253A12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew%253B%2520example%253Akey%253Did%253Dbafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%2522%252C%250A%2520%2520%2520%2520%2522signatureValue%2522%253A%2520%2522o9r6LxgoGN8FoaeeUA6EdDcv12GvDzFEmCgjWzvpur2YSQyA8W2r0SSWUK%252BnH5tMqzaFLun6wwZ1Eot37amGDg%253D%253D%2522%252C%250A%2520%2520%2520%2520%2522type%2522%253A%2520%2522ed25519Signature2018%2522%250A%2520%2520%257D%252C%250A%2520%2520%2522verificationMethod%2522%253A%2520%255B%250A%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522curve%2522%253A%2520%2522ed25519%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522expires%2522%253A%2520%25222019-12-01T03%253A00%253A00Z%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522id%2522%253A%2520%2522bafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%253Bexample%253Akey%253Didbafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522publicKeyBase64%2522%253A%2520%2522qmz7tpLNKKKdl7cD7PbejDiBVp7ONpmZbfmc7cEK9mg%253D%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522type%2522%253A%2520%2522EdDsaPublicKey%2522%250A%2520%2520%2520%2520%257D%250A%2520%2520%255D%252C%250A%2520%2520%2522updated%2522%253A%2520%25222020-05-01T03%253A00%253A00Z%2522%250A%257D&tab=0)

***
<a href="https://pr-preview.s3.amazonaws.com/gjgd/did-core/pull/430.html" title="Last updated on Oct 26, 2020, 7:07 PM UTC (fb9e218)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/430/e03c728...gjgd:fb9e218.html" title="Last updated on Oct 26, 2020, 7:07 PM UTC (fb9e218)">Diff</a>

### CODE REVIEW

**Comment by OR13** [comment]
@gjgd `proof` was removed, if an example uses it, it needs to include https://w3id.org/security/v2 as well.

**Comment by OR13** [comment]
lol, nvm i see you fixed this already.

File: index.html [hunk 1]
```diff
@@ -2825,7 +2825,10 @@ <h4>DagCBOR</h4>
           </ul>
 
           <pre class="example nohighlight" title="DID Document as DagCBOR same as the previous example, but serialized to JSON for easy readability">
-{ "@context": "https://www.w3.org/ns/did/v1",
+{ "@context": [
+    "https://www.w3.org/ns/did/v1",
+    "https://w3id.org/security/v2"
+  ],
```
Comment (msporny) [inline index.html:2831]: Hrm, we shouldn't pull in https://w3id.org/security/v2 -- the terms that are missing from the vocabulary should be defined in did/v1 -- do you know what was missing @gjgd?

**Review by msporny — CHANGES_REQUESTED** [review]
We should fix the the did/v1 vocab... not add the security vocab.

**Comment by OR13** [comment]
@msporny we removed `proof` from the did vocabulary... We can either change the example (so it does not have a property not defined in the context), or this PR is correct.

File: index.html (hunk 1 above)
Comment (OR13) [reply index.html:2831]: `proof`

Comment (gjgd) [reply index.html:2831]: Screenshot 2020-10-08 at 15 57 20

From [json ld checker](https://www.jsonld-checker.com/?json=%257B%250A%2520%2520%2522%2540context%2522%253A%2520%2522https%253A%252F%252Fwww.w3.org%252Fns%252Fdid%252Fv1%2522%252C%250A%2520%2520%2522authentication%2522%253A%2520%255B%250A%2520%2520%2520%2520%2522did%253Aexample%253A12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew%253Bkey-id%253Dbafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%2522%250A%2520%2520%255D%252C%250A%2520%2520%2522created%2522%253A%2520%25222018-12-01T03%253A00%253A00Z%2522%252C%250A%2520%2520%2522id%2522%253A%2520%2522did%253Aexample%253A12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew%2522%252C%250A%2520%2520%2522proof%2522%253A%2520%257B%250A%2520%2520%2520%2520%2522created%2522%253A%2520%25222020-05-01T03%253A00%253A02Z%2522%252C%250A%2520%2520%2520%2520%2522creator%2522%253A%2520%2522did%253Aexample%253A12D3KooWMHdrzcwpjbdrZs5GGqERAvcgqX3b5dpuPtPa9ot69yew%253B%2520example%253Akey%253Did%253Dbafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%2522%252C%250A%2520%2520%2520%2520%2522signatureValue%2522%253A%2520%2522o9r6LxgoGN8FoaeeUA6EdDcv12GvDzFEmCgjWzvpur2YSQyA8W2r0SSWUK%252BnH5tMqzaFLun6wwZ1Eot37amGDg%253D%253D%2522%252C%250A%2520%2520%2520%2520%2522type%2522%253A%2520%2522ed25519Signature2018%2522%250A%2520%2520%257D%252C%250A%2520%2520%2522verificationMethod%2522%253A%2520%255B%250A%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522curve%2522%253A%2520%2522ed25519%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522expires%2522%253A%2520%25222019-12-01T03%253A00%253A00Z%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522id%2522%253A%2520%2522bafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%253Bexample%253Akey%253Didbafyreicubtx5wqo3nosc4cazrkctfhwd6rewezgpwoe4swirls4ebdhs2i%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522publicKeyBase64%2522%253A%2520%2522qmz7tpLNKKKdl7cD7PbejDiBVp7ONpmZbfmc7cEK9mg%253D%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522type%2522%253A%2520%2522EdDsaPublicKey%2522%250A%2520%2520%2520%2520%257D%250A%2520%2520%255D%252C%250A%2520%2520%2522updated%2522%253A%2520%25222020-05-01T03%253A00%253A00Z%2522%250A%257D&tab=0)

Comment (msporny) [reply index.html:2831]: Doh! I think this snuck in there with the DAG CBOR PR -- agreed, @gjgd, the right thing to do here is remove `proof`. /cc @jonnycrunch -- please make sure you don't include `proof` in future examples. It was removed from the spec a while ago.

Comment (jonnycrunch) [reply index.html:2831]: removal of proof came as a surprise to me.  [Recent PR](https://github.com/w3c/did-core/pull/420) with updated CBOR section it has been removed.

Comment (gjgd) [reply index.html:2831]: Will do, though I think we also need to remove `verificationMethod`. I'm not an expert in DagCBOR, would the resulting document (without proof and verificationMethod) still illustrate [the corresponding section](https://w3c.github.io/did-core/#dagcbor) @msporny ?

Comment (OR13) [reply index.html:2831]: See https://github.com/w3c/did-spec-registries/blob/master/contexts/did-v1.jsonld, IMO, the example should match terms that are defined in that file.

**Comment by msporny** [comment]
> @msporny we removed `proof` from the did vocabulary... We can either change the example (so it does not have a property not defined in the context), or this PR is correct.

I don't think the example depends on `proof` to be there, does it? Injecting the security context in an example is strange, IMHO. We should remove `proof` from the example. @gjgd, @OR13  -- do you agree? If so, could you please revise this PR so we can merge it in?

**Comment by OR13** [comment]
@gjgd please remove any properties from examples that are not in the context.

**Comment by gjgd** [comment]
@msporny agreed, I removed the proof property.

The other issue still remains
> - [The UserHubEndpoint object](https://www.jsonld-checker.com?tab=0&jsonid=bc55eabd-6c26-47bb-be93-920bff02cf7e) which context (https://schema.identity.foundation/hub) is not a resolvable URL. We should either remove this object or fix it by adding a valid context. I wasn't able to find a context with these properties

Any suggestions on where to find the context for this object?

**Comment by peacekeeper** [comment]
> Any suggestions on where to find the context for this object?

Not sure if this context exists yet, but the authoritative place where it should be defined is probably https://identity.foundation/secure-data-store/#did-document-service-endpoint-descriptors.

**Review by msporny — APPROVED** [review]

**Comment by msporny** [comment]
Editorial, multiple reviews, changes requested and made, merging.

**Pull request merged**